src/components/App.js

```diff
@@ -1,16 +1,163 @@
+    this.clearOrder = this.clearOrder.bind(this);
+    this.getOrderHistory = this.getOrderHistory.bind(this);
+    this.handleClearHistory = this.handleClearHistory.bind(this);
+    this.closeWasClicked = this.closeWasClicked.bind(this);
+  }
+
+  returnMenu(result) {
+    const resultArr = Object.keys(result).map(function(item) {
+      return result[item];
+    });
+    this.setState({
+      menu: resultArr
+    });
+  }
+
+  getOrder(order) {
+    const things = Array.from(this.state.orders);
+    const matchingOrder = things.find(currentOrder => {
+      return order.id === currentOrder.id;
+    });
+
+    if (matchingOrder) {
+      matchingOrder.quantity += order.quantity;
+      matchingOrder.price += order.price;
+
+      const ordersWithoutCurrentOrder = this.state.orders.filter(
+        currentOrder => {
+          return currentOrder.id !== order.id;
+        }
+      );
+
+      this.setState({
+        orders: ordersWithoutCurrentOrder.concat(matchingOrder)
+      });
+    } else {
+      this.setState({
+        orders: [...this.state.orders, order]
+      });
+    }
+  }
+
+  handleDelete(key) {
+    this.setState(prevState => ({
+      orders: prevState.orders.filter(el => el.orderId != key)
+    }));
+  }
+
+  clearOrder() {
+    this.setState({
+      orders: {}
+    });
+  }
+
+  getOrderHistory(event) {
+    event.preventDefault();
+    fetch("/api/order", { method: "get" })
+      .then(response => response.json())
+      .then(result => {
+        if (Object.keys(result).length === 0) {
+          alert(
+            "โปรดสั่งซื้อเพื่อดูประวัติคำสั่งซื้อ || Please place an order to view order history"
+          );
+        } else {
+          this.setState({
+            orderhistory: result,
+            closed: false
+          });
+        }
+      })
+      .catch(error => console.log(error));
+  }
+
+  handleClearHistory(key) {
+    // const tempOrderHistory = [...this.state.orderhistory];
+    // delete tempOrderHistory[key];
+    // this.setState({
+    //   orderhistory: tempOrderHistory
+    // });
+
+    fetch(`/api/order/${key}`, { method: "delete" })
+      .then(response => response.json())
+      .then(result => {
+        console.log("delete result:", result);
+        this.setState({
+          orderhistory: result
+        });
+      })
+      .catch(error => console.log(error));
+  }
+
+  closeWasClicked() {
+    this.setState({
+      closed: true
+    });
   }
 
-  render(){
+  render() {
+    console.log("set order hist:", this.state.orderhistory);
     return (
-      <div>
-        Delivereat app
+      <div className="app">
+        <div className="header">
+          <Nav />
+        </div>
+        <button
+          className="orderform__history button"
+          onClick={this.getOrderHistory}
+          type="submit"
+        >
+          View Order History
+        </button>
+        <div className="main">
+          {Object.keys(this.state.orderhistory).length === 0 ||
```

Comment:
This could be written slightly shorter as `!Object.keys(this.state.orderhistory).length` since if length is 0, it will be falsy